src/prefect_server/api/runs.py

```diff
@@ -419,8 +421,28 @@ async def delete_flow_run(flow_run_id: str) -> bool:
     if not flow_run_id:
         raise ValueError("Invalid flow run ID.")
 
-    result = await models.FlowRun.where(id=flow_run_id).delete()
-    return bool(result.affected_rows)  # type: ignore
+    # Delete task run states and task runs first to speedup deletion which can be
+    # _very_ slow on the large task tables due since they are implemented as
+    # row-triggers and the foreign key checks are slow
+
+    client = HasuraClient()
```

Comment:
You'll need to use `prefect.plugins.hasura.client` here so that when this route is called from within Cloud, the right client is used.

Reply:
@cicdw This route isn't called from `Cloud`, it has its own `delete_flow_run` api call

Reply:
Yea, I saw that but would rather not bank on it just in case it gets removed for any reason